### ISSUE TITLE
Lookup and indexing functions

### ISSUE THREAD

**Issue by ocramz** [andrewthad/colonnade#29]
Is there a way to lookup indices, filter rows and so on?

**Comment by andrewthad** [comment]
Can you explain what you mean? I think that, to filter rows, you would filter the collection that you are encoding it before passing it to any of colonnade’s function.

**Comment by ocramz** [comment]
Yes sorry for the brief ticket, this confirms my understanding of the API.

The internal OneColonnade ("Column" ?) type is a sort of "pull array" [1] but has no notion of row index. 

I was looking for a way to use 'colonnade' as the basis for a "dataframe" of sorts, like Apache Arrow.

1) https://hackage.haskell.org/package/massiv-1.0.0.0/docs/src/Data.Massiv.Array.Delayed.Pull.html#DArray
